package-res/reportviewer/reportviewer-prompt.js

```diff
@@ -93,7 +93,7 @@ define([
               // add to changed
               changedParameters.push(parameter.name);
             }
-          } else {
+          } else if("" != parameter.value) {
```

Comment:
At this point I am testing the following situation:
Comparing _oldParameterSet and currentParameterSet. 
currentParameterSet is a result of operation.getParameterValues() so it's not contains parameters with empty values.
_oldParameterSet contains empty parameters, so here I'm checking:
if currentParameterSet has such parameter AND _oldParameterSet has not such parameter AND parameter values is not empty - push it to the changedParameters.